pkg/auth/requests/authenticate.go

```diff
@@ -81,31 +92,37 @@ func tokenKeyIndexer(obj interface{}) ([]string, error) {
 	return []string{token.Token}, nil
 }
 
-func (a *tokenAuthenticator) Authenticate(req *http.Request) (bool, string, []string, error) {
+func (a *tokenAuthenticator) Authenticate(req *http.Request) (*AuthenticatorResponse, error) {
```

Comment:
if the return is not a pointer then the errors below can return nil without having to create an object

Reply:
they can return nil even when  return is a pointer, right?